### PR TITLE
[Chat] - IdentifierModels - Moving the models into chat

### PR DESCRIPTION
**Pull request by JoshuaLai**

### CODE REVIEW

File: sdk/communication/AzureCommunication/Tests/CommunicationIdentifierSerializerTests.swift [hunk 1]
```diff
@@ -132,69 +189,25 @@ class CommunicationIdentifierSerializerTests: XCTestCase {
...
-                rawId: expectedId
-            ))
-
-        XCTAssertEqual(model.kind, .microsoftTeamsUser)
-        XCTAssertEqual(model.microsoftTeamsUserId, "user id")
-        XCTAssertEqual(model.isAnonymous, isAnonymous)
-        XCTAssertEqual(model.id, expectedId)
-    }
-
-    func test_DeserializeMissingProperty() throws {
```
Comment (RezaJooyandeh) [inline sdk/communication/AzureCommunication/Tests/CommunicationIdentifierSerializerTests.swift:156]: why is this test not needed anymore?

File: sdk/communication/AzureCommunicationChat/Source/Generated/Models/MicrosoftTeamsUserIdentifierModel.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+import AzureCore
+import Foundation
+// swiftlint:disable superfluous_disable_command
+// swiftlint:disable identifier_name
+// swiftlint:disable line_length
+// swiftlint:disable cyclomatic_complexity
+
+/// A Microsoft Teams user.
+public struct MicrosoftTeamsUserIdentifierModel: Codable, Equatable {
```
Comment (RezaJooyandeh) [inline sdk/communication/AzureCommunicationChat/Source/Generated/Models/MicrosoftTeamsUserIdentifierModel.swift:19]: let's make sure the model types are not public

Comment (JoshuaLai) [reply sdk/communication/AzureCommunicationChat/Source/Generated/Models/MicrosoftTeamsUserIdentifierModel.swift:19]: The `public` was automatically added in the generated file. Should this be removed (we would need to manually do it though)

File: sdk/communication/AzureCommunication/Tests/CommunicationIdentifierSerializerTests.swift (hunk 1 above)
Comment (JoshuaLai) [reply sdk/communication/AzureCommunication/Tests/CommunicationIdentifierSerializerTests.swift:156]: Screen Shot 2021-02-11 at 5 42 46 PM
Since the new init has a bunch of default parameters technically these tests for the model become less relevant/ not possible.

Comment (RezaJooyandeh) [reply sdk/communication/AzureCommunication/Tests/CommunicationIdentifierSerializerTests.swift:156]: but some of the combinations that we want to prohibit are valid in the schema. For example, in deserialization we want to enforce `rawId` to be present. In this example `rawId` is nullable because it is not needed for serialization.

File: sdk/communication/AzureCommunicationChat/Source/Generated/Models/MicrosoftTeamsUserIdentifierModel.swift (hunk 2 above)
Comment (RezaJooyandeh) [reply sdk/communication/AzureCommunicationChat/Source/Generated/Models/MicrosoftTeamsUserIdentifierModel.swift:19]: In some other languages there are ways to inform the code generator to generate as non public. I am wondering if autorest.swift already has mechanisms for it.

File: sdk/communication/AzureCommunication/Tests/CommunicationIdentifierSerializerTests.swift (hunk 1 above)
Comment (JoshuaLai) [reply sdk/communication/AzureCommunication/Tests/CommunicationIdentifierSerializerTests.swift:156]: Oh I see your point. Ok yes I will need to update these tests!

File: sdk/communication/AzureCommunicationChat/Source/Generated/Models/MicrosoftTeamsUserIdentifierModel.swift (hunk 2 above)
Comment (JoshuaLai) [reply sdk/communication/AzureCommunicationChat/Source/Generated/Models/MicrosoftTeamsUserIdentifierModel.swift:19]: Had a quick chat with @tjprescott and looks like we don't have this mechanism for swift autorest

File: sdk/communication/AzureCommunication/Tests/CommunicationIdentifierSerializerTests.swift (hunk 1 above)
Comment (JoshuaLai) [reply sdk/communication/AzureCommunication/Tests/CommunicationIdentifierSerializerTests.swift:156]: @RezaJooyandeh re-introduced the these tests in the `Chat/IdentifierSerializerTests`

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,10 @@
 ## 1.0.0-beta.9 (Unreleased)
 ### New Features
 
+### New Features
+- Azure Communication Common Library
+ - Introduction of  a new struct `CommunicationIdentifierModel` to repesent a union type that is either a `communicationUser`, `phoneNumber`, or `microsoftTeamsUser`.  Along side this new object we introduced `CommunicationIdentifierSerializer` to serialize and deserialize valid JSON.
```
Comment (RezaJooyandeh) [inline CHANGELOG.md:8]: Are not we moving this type to the Chat package? I am also not sure if we should add this here. The type is only public because autorest.swift can't make it internal at this stage; otherwise this type is for internal usage only. @tjprescott , do you have suggestions on if we should include this in the release notes or not.

Comment (tjprescott) [reply CHANGELOG.md:8]: I would say remove it. Is the intent that both the common and the Chat SDKs will have this model internally or is Chat supposed to leverage the implementation in Common? If it's the second case, the the model must be public.

Comment (JoshuaLai) [reply CHANGELOG.md:8]: The model is going to live in the Chat package. Would it be better instead of having it under `Azure Communication Common Library` to update it to be under the `Azure Communication Chat Library`? Because this change log still does technically apply @RezaJooyandeh @tjprescott

File: sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift [hunk 4]
```diff
@@ -39,22 +39,15 @@ import Foundation
...
 
-    public static func == (lhs: CommunicationCloudEnvironment, rhs: CommunicationCloudEnvironment) -> Bool {
+    public static func isEqual(lhs: CommunicationCloudEnvironment, rhs: CommunicationCloudEnvironment) -> Bool {
         if lhs.environmentValue == rhs.environmentValue {
             return true
         } else {
             return false
         }
     }
 }
```
Comment (tjprescott) [inline sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift:53]: Why this change?  The `==` operator is the idiomatic implementation.

File: sdk/communication/AzureCommunicationChat/Source/Generated/Models/ChatMessageReadReceipt.swift [hunk 5]
```diff
@@ -16,7 +16,7 @@ import Foundation
 // swiftlint:disable cyclomatic_complexity
 
 /// A chat message read receipt indicates the time a chat message was read by a recipient.
-public struct ChatMessageReadReceipt: Codable {
+public struct ChatMessageReadReceipt: Codable, Equatable {
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationChat/Source/Generated/Models/ChatMessageReadReceipt.swift:19]: Is this regenerated or manually added?

File: sdk/core/AzureCore/Source/Pipeline/Policies/ContentDecodePolicy.swift [hunk 6]
```diff
@@ -224,7 +224,8 @@ public class ContentDecodePolicy: PipelineStage {
             contentType: contentType,
```
Comment (tjprescott) [inline sdk/core/AzureCore/Source/Pipeline/Policies/ContentDecodePolicy.swift:224]: This should be reverted.

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,10 @@
 ## 1.0.0-beta.9 (Unreleased)
 ### New Features
 
+### New Features
```
Comment (tjprescott) [inline CHANGELOG.md:6]: Duplicate line. Remove one

**Review by tjprescott — CHANGES_REQUESTED** [review]
Please revert any changes to AzureCore.

File: sdk/communication/AzureCommunicationChat/Source/Generated/Models/ChatMessageReadReceipt.swift (hunk 5 above)
Comment (JoshuaLai) [reply sdk/communication/AzureCommunicationChat/Source/Generated/Models/ChatMessageReadReceipt.swift:19]: This was generated from autorest

File: CHANGELOG.md (hunk 3 above)
Comment (tjprescott) [reply CHANGELOG.md:6]: Well, it should be in both--removed from AzureCommunication and added to Chat. It would be a breaking change for AzureCommunication, yes?

**Review by tjprescott — APPROVED** [review]
